Build fail between 1.1a8 to 1.6.5 on Debian stretch(9)

I countered build fails with 1.1a8 to 1.6.5 with following error message.

```
libruby.a(parse.o): In function `rb_yylex':
/all-ruby/1.6.5/ruby-1.6.5/parse.y:3620: undefined reference to `rb_reserved_word'
collect2: error: ld returned 1 exit status
```

--- a/Rakefile
+++ b/Rakefile
@@ -542,7 +542,7 @@ class RubySource
     %w[gcc cc].each {|cc_bin|
       File.open("#{prefix}/bin/#{cc_bin}", "w") {|f|
         f.puts "#!/bin/sh"
-        f.puts "#{gcc} -m32 \"$@\""
+        f.puts "#{gcc} -m32 -std=gnu89 \"$@\""
       }
       File.chmod(0755, "#{prefix}/bin/#{cc_bin}")
     }

